### PR TITLE
Refine BBF_GC_SAFE_POINT handling in reachability computation

### PR DESCRIPTION
**Pull request by BruceForstall**

### CODE REVIEW

**Comment by BruceForstall** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=225391&view=ms.vss-build-web.run-extensions-tab)

No asm diffs (as expected).

TP improvement on 64-bit of 0.01% to 0.02%. TP regression on 32-bit of 0.01%. Presumably because `BasicBlockFlags` is `unsigned __int64`

**Comment by BruceForstall** [comment]
TP diffs slightly better than https://github.com/dotnet/runtime/pull/84220 option.

**Comment by BruceForstall** [comment]
@AndyAyersMS Worth a slight 64-bit TP improvement for a slight 32-bit TP regression?

File: src/coreclr/jit/fgopt.cpp [hunk 1]
```diff
@@ -260,20 +260,17 @@ void Compiler::fgComputeReachabilitySets()
...
 
-                if (!(predBlock->bbFlags & BBF_GC_SAFE_POINT))
-                {
-                    predGcSafe = false;
-                }
+                predGcFlags &= predBlock->bbFlags;
             }
 
-            if (predGcSafe)
+            if ((block->bbPreds != nullptr) && (predGcFlags != BBF_EMPTY))
```
Comment (AndyAyersMS) [inline src/coreclr/jit/fgopt.cpp:273]: Avoid a few more branches?

```suggestion
            block->bbFlags |= predGcFlags;
```

**Review by AndyAyersMS — COMMENTED** [review]
Also strikes me as odd that we don't keep iterating if we update the block flags. 

Perhaps it doesn't matter and we get the same answers as if we kept going. But it's not obvious to me that this happens. Seems like we could give up too early and under-mark the set of blocks that are GC safe.

**Comment by BruceForstall** [comment]
> Also strikes me as odd that we don't keep iterating if we update the block flags.

I thought about that, but it's not a correctness matter because if we don't mark all the potential GC safe blocks, we'll fall back to fully interruptible code. Given how much reachability iterates, it might converge anyway (at least most of the time).

File: src/coreclr/jit/fgopt.cpp (hunk 1 above)
Comment (BruceForstall) [reply src/coreclr/jit/fgopt.cpp:273]: We have to check `block->bbPreds != nullptr` or else we'd set any block with no predecessors (commonly the first block) with `BBF_GC_SAFE_POINT` unconditionally. But true, we don't need to check against `BBF_EMPTY`.

**Review by AndyAyersMS — APPROVED** [review]

**Comment by BruceForstall** [comment]
[Latest diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=225532&view=ms.vss-build-web.run-extensions-tab)

TP improvement (-0.03% to -0.01%) on 64-bit, no change on 32-bit.

**Pull request merged**